src/to_wkt/geo_trait_impl.rs

```diff
@@ -0,0 +1,480 @@
+            write_multi_polygon(f, multi_polygon)
+        }
+        geo_traits::GeometryType::GeometryCollection(gc) => write_geometry_collection(f, gc),
+        geo_traits::GeometryType::Rect(rect) => write_rect(f, rect),
+        geo_traits::GeometryType::Triangle(triangle) => write_triangle(f, triangle),
+        geo_traits::GeometryType::Line(line) => write_line(f, line),
+    }
+}
+
+pub fn write_geometry_collection<T: WktNum + fmt::Display, G: GeometryCollectionTrait<T = T>>(
+    f: &mut impl Write,
+    gc: &G,
+) -> Result<(), Error> {
+    let dim = gc.dim();
+    // Write prefix
+    match dim {
+        geo_traits::Dimensions::Xy | geo_traits::Dimensions::Unknown(2) => {
+            f.write_str("GEOMETRYCOLLECTION")
+        }
+        geo_traits::Dimensions::Xyz | geo_traits::Dimensions::Unknown(3) => {
+            f.write_str("GEOMETRYCOLLECTION Z")
+        }
+        geo_traits::Dimensions::Xym => f.write_str("GEOMETRYCOLLECTION M"),
+        geo_traits::Dimensions::Xyzm | geo_traits::Dimensions::Unknown(4) => {
+            f.write_str("GEOMETRYCOLLECTION ZM")
+        }
+        geo_traits::Dimensions::Unknown(_) => return Err(Error::UnknownDimension),
+    }?;
+    let mut geometries = gc.geometries();
+
+    if let Some(first_geometry) = geometries.next() {
+        f.write_str("(")?;
+
+        write_geometry(f, &first_geometry)?;
+        for geom in geometries {
+            f.write_char(',')?;
+            write_geometry(f, &geom)?;
+        }
+
+        f.write_char(')')?;
+    } else {
+        f.write_str(" EMPTY")?;
+    }
+    Ok(())
+}
+
+pub fn write_rect<T: WktNum + fmt::Display, G: RectTrait<T = T>>(
+    f: &mut impl Write,
+    rect: &G,
+) -> Result<(), Error> {
+    // Write prefix and error if not 2D
+    match rect.dim() {
+        geo_traits::Dimensions::Xy | geo_traits::Dimensions::Unknown(2) => f.write_str("POLYGON"),
+        _ => return Err(Error::RectUnsupportedDimension),
+    }?;
+
+    let min_coord = rect.min();
+    let max_coord = rect.max();
+
+    // We need to construct the five points of the rect that make up the exterior Polygon
+    let coords = [
+        Coord {
+            x: min_coord.x(),
+            y: min_coord.y(),
+            z: None,
+            m: None,
+        },
+        Coord {
+            x: min_coord.x(),
+            y: max_coord.y(),
+            z: None,
+            m: None,
+        },
+        Coord {
+            x: max_coord.x(),
+            y: max_coord.y(),
+            z: None,
+            m: None,
+        },
+        Coord {
+            x: max_coord.x(),
+            y: min_coord.y(),
+            z: None,
+            m: None,
+        },
+        Coord {
+            x: min_coord.x(),
+            y: min_coord.y(),
+            z: None,
+            m: None,
+        },
+    ];
+
+    f.write_str("(")?;
+    write_coord_sequence(f, coords.iter(), PhysicalCoordinateDimension::Two)?;
+    Ok(f.write_char(')')?)
+}
```

Comment:
Could be as well be direct like:

```
write!(
        f,
        "POLYGON ({0} {1},{2} {1},{2} {3},{0} {3},{0} {1})",
        min.x(),
        min.y(),
        max.x(),
        max.y(),
 )?;
```

It was not materialized at the time it was copied; sorry for that.